wall_e/src/WalleModels/models.py

```diff
@@ -6,6 +6,73 @@
 from django.conf import settings
 from django.db import models
 from django.forms import model_to_dict
+from .customFields import FixedCharField, GeneratedIdentityField
+import logging
+logger = logging.getLogger('wall_e')
+
+
+class BanRecords(models.Model):
+    ban_id = GeneratedIdentityField(primary_key=True, always=True)
+    username = models.CharField(max_length=32, null=False)
+    user_id = FixedCharField(max_length=18, null=False)
+    mod = models.CharField(max_length=32, null=True)
+    mod_id = FixedCharField(max_length=18, null=True)
+    date = models.DateTimeField(unique=True, null=True)
+    reason = models.TextField(null=False)
+
+    class Meta:
+        unique_together = ['user_id', 'date']
+        db_table = 'WalleModels_ban_records'
+
+    @classmethod
+    @sync_to_async
+    def insert_record(cls, username, user_id, mod, mod_id, date, reason):
+        """Adds entry to BanRecords table"""
+
+        logger.info("[BanRecords insert_record()] Adding ban record to with values" +
+                    f"({username, user_id, mod, mod_id, date, reason})")
+
+        BanRecords(username=username, user_id=str(user_id), mod=mod, mod_id=str(mod_id), date=date,
+                   reason=reason).save()
+
+
+class BannedUsers(models.Model):
+    username = models.CharField(max_length=32, null=False)
+    user_id = FixedCharField(primary_key=True, max_length=18)
+
+    class Meta:
+        db_table = 'WalleModels_banned_users'
+
+    @classmethod
+    @sync_to_async
+    def insert_ban(cls, username, user_id):
+        """Adds entry to BannedUsers table"""
+
+        logger.info(f"[BannedUsers insert_ban()] Adding banned user to with values ({username, user_id})")
+        BannedUsers(username, str(user_id)).save()
+
+    @classmethod
+    @sync_to_async
+    def get_banned_ids(cls):
+        """Gets list of all user_ids from BannedUsers table"""
+
+        return list(map(int, (BannedUsers.objects.values_list('user_id', flat=True))))
+
+    @classmethod
+    @sync_to_async
+    def del_banned_user_by_id(cls, _id):
+        """Deletes single entry from BannedUsers by user_id """
+
+        user = BannedUsers.objects.filter(user_id=str(_id))[0]
+        name = user.username
+        user.delete()
+        return name
+
+    @classmethod
+    @sync_to_async
+    def get_all_bans(cls):
+        """Gets list of all entries in BannedUsers"""
+        return list(BannedUsers.objects.values_list('username', 'user_id'))
```

Comment:
why not just do `return list(BannedUsers.objects.all())`?

this way you don't need to do array indexing on the results.